### PR TITLE
Add grouping support for ray CI pipeline.

### PR DESCRIPTION
**Pull request by aslonnie**
So that we can support grouping.

One can add a `#ci:group=MY_GROUP_NAME` to put all the steps into a group with name `MY_GROUP_NAME`.

### CODE REVIEW

File: ray_ci/pipeline_ci.py [hunk 1]
```diff
@@ -294,7 +303,13 @@ def _print_command(cmd: str) -> List[str]:
...
 
     # Print to stdout
-    steps_str = json.dumps(pipeline_steps)
+    if group_name:
+        pipeline_steps = {
+            "group": group_name,
+            "steps": pipeline_steps,
+        }
+
+    steps_str = json.dumps({"steps:": pipeline_steps})
```
Comment (aslonnie) [inline ray_ci/pipeline_ci.py:312]: based on buildkite official docs, steps should have a `steps` as key, rather than just a raw list.

**Comment by aslonnie** [comment]
I did run the pytest locally.. probably should set a ci for the ci..

**Comment by can-anyscale** [comment]
Should the pipeline yaml file be written in a syntax that buildkite itself can interpret as a group, like:

steps:
  - group:
     steps:
       - ...

then we don't have to support a special syntax for grouping?

**Comment by aslonnie** [comment]
> Should the pipeline yaml file be written in a syntax that buildkite itself can interpret as a group

I thought about this, but that would need to add an extra indentation to all the build yaml files, and will destroy all the line-based blaming.

where this comment based syntax only needs to add one new line.

so.. what do you think we keep it like this?

**Comment by aslonnie** [comment]
Or I can add magic step entry that can have the group name and get dropped during the parsing, so that it is part of the yaml. let me know which one you prefer.

**Comment by aslonnie** [comment]
ping @can-anyscale ; any opinion / feedback ?

**Comment by can-anyscale** [comment]
Oh, I wanted to ask what the magic step entry is but I forgot. But I think the current solution is also fine on its own.

**Review by can-anyscale — APPROVED** [review]

**Comment by aslonnie** [comment]
> I wanted to ask what the magic step entry is but I forgot

like:

````
- default_group: build
- label: print hello
  command:
  - echo hello
- label: build
  commands:
  - bazel build ...
````

and the script filters out `default_group` from the list and pick it up as a group name.

**Comment by aslonnie** [comment]
if you like the magic step style, I can also implement that instead.

the magic comment style is kind of stolen from go:

https://pkg.go.dev/go/build#hdr-Build_Constraints

**Comment by aslonnie** [comment]
I think I just broke everything..

**Pull request merged**